fix: enhance existing authentication docs (#1780)
pypa/hatch: docs/how-to/publish/auth.md

----

> To publish to an index such as PyPI or TestPyPI you need two things:

Reviewer: ```suggestion
To publish to a package index such as PyPI or TestPyPI you need two things:
```

> 1. a username (e.g. `__token__` for PyPI)

Reviewer: ```suggestion
1. a username (e.g. `__token__` for PyPI to select two-factor authentication aka. 2FA)
```

> 2. a token value or other form of trusted authentication from the index (PyPI)

Reviewer: ```suggestion
2. a password, token value or other form of trusted authentication for the package index
```
